vault/external_tests/consul_fencing_binary/consul_fencing_test.go

```diff
@@ -263,6 +275,46 @@ func TestConsulFencing_PartitionedLeaderCantWrite(t *testing.T) {
 	require.Equal(t, expect1, sets[1], "Client 1 writes lost")
 }
 
+func waitForKVv2Upgrade(t *testing.T, ctx context.Context, client *api.Client, path string) {
+	t.Helper()
+
+	ctx, cancel := context.WithTimeout(ctx, 10*time.Second)
+	defer cancel()
+
+	kv := client.KVv2(path)
+	attempts := 0
+	wait := 20 * time.Millisecond
+	for {
+		// Attempt to perform a write on the KVv2 mount. It will fail until the
+		// backend is done upgrading. If this is a performance standby in Ent then
+		// it will not complete until the primary is done upgrading AND the standby
+		// has noticed that!
+		_, err := kv.Put(ctx, "test-upgrade-done", map[string]interface{}{
+			"ok": 1,
+		})
+		if err == nil {
+			return
+		}
+		t.Logf("waitForKVv2Upgrade: write faile: %s", err)
```

Comment:
it might be worth checking the error to verify that it is an upgrade error, and erroring out if it's something else.

Reply:
also, typo for the word `faile`

Reply:
Do we need this? Maybe we could just include an aggregate of all the error messages to the fatal message in `ctx.Done` case?

Reply:
I don't feel strongly about it, I'm fine to merge it as is

Reply:
@miagilepner I think I'm leaning towards merging this as it is (with typo fix) and we can see if the logs are noisy or it takes too long to fail for other reasons. My hunch is the timeout is short enough and the backoff sufficient that it won't be too noisy or problematic like this and will be more robust to changes. If you feel strongly though I'm happy to go back!